an/v1/AN_v1.0.0-Beta-1.yaml

```diff
@@ -0,0 +1,2464 @@
+        - $ref: '#/components/schemas/UNLocationLocation'
+        - $ref: '#/components/schemas/FacilityLocation'
+      discriminator:
+        propertyName: locationType
+        mapping:
+          ADDR: '#/components/schemas/AddressLocation'
+          UNLO: '#/components/schemas/UNLocationLocation'
+          FACI: '#/components/schemas/FacilityLocation'
+
+    #################
+    # Return Location
+    #################
+    ReturnLocation:
+      title: Return Location
+      type: object
+      description: |
+        General purpose object to capture `Return` location specified as: specific equipment handling facility where container is to be returned by the consignee or its logistics partner.
+
+        The location can be specified in **one** of the following ways: `UN Location Code`, `Facility` or an `Address`.
+
+        **Condition:** only applicable if there are Carrier Owned Containers (no SOC).
+      example:
+        locationName: Hamburg
+        locationType: UNLO
+        UNLocationCode: DEHAM
+      oneOf:
+        - $ref: '#/components/schemas/AddressLocation'
+        - $ref: '#/components/schemas/UNLocationLocation'
+        - $ref: '#/components/schemas/FacilityLocation'
+      discriminator:
+        propertyName: locationType
+        mapping:
+          ADDR: '#/components/schemas/AddressLocation'
+          UNLO: '#/components/schemas/UNLocationLocation'
+          FACI: '#/components/schemas/FacilityLocation'
+
+    ############################
+    # Address Location Interface
+    ############################
+    AddressLocation:
+      title: Address Location
+      type: object
+      description: An interface used to express a location using an `Address` object.
+      properties:
+        locationName:
+          type: string
+          description: The name of the location.
+          example: Port of Amsterdam
+          maxLength: 100
+        locationType:
+          type: string
+          maxLength: 4
+          description: |
+            Discriminator used to identify this as an `Address` location interface.
+          example: ADDR
+        address:
+          $ref: '#/components/schemas/Address'
+      required:
+        - locationType
+        - address
+    Address:
+      type: object
+      title: Address
+      description: |
+        An object for storing address related information
+      properties:
+        street:
+          type: string
+          maxLength: 100
+          description: The name of the street of the party’s address.
+          example: Ruijggoordweg
+        streetNumber:
+          type: string
+          maxLength: 50
+          description: The number of the street of the party’s address.
+          example: '100'
+        floor:
+          type: string
+          pattern: '^\S(?:.*\S)?$'
+          maxLength: 50
+          description: |
+            The floor of the party’s street number.
+          example: 2nd
+        postCode:
```

Comment:
Should the PostCode be of maxlength 50 or 10? In IFS it is mentioned 10.

Reply:
Good spotted - this is copied from EBL and Booking where it is also wrong. I will make an issue to fix in all APIs